src/transaction/TransactionController.ts

```diff
@@ -541,7 +543,7 @@ export class TransactionController extends BaseController<
     });
 
     transactions.push(transactionMeta);
-    this.update({ transactions: [...transactions] });
+    this.trimTransactionsForState(transactions);
```

Comment:
I think rather than adding a new method it would be easier to grok if it were something like:
```js
this.update({ transactions: trimTransactions(transactions) });
```
wdyt?

Reply:
Good suggestion. Updated PR